data/api/client-server/report_content.yaml

```diff
@@ -16,6 +16,69 @@ info:
   title: Matrix Client-Server Report Content API
   version: 1.0.0
 paths:
+  "/rooms/{roomId}/report":
+    post:
+      summary: Report a room as inappropriate.
+      description: |-
+        Reports a room as inappropriate to the server, which may then notify
+        the appropriate people. The caller is not required to be joined to the
+        room to report it.
+      operationId: reportRoom
+      parameters:
+        - in: path
+          name: roomId
+          description: The room being reported.
+          required: true
+          example: "!637q39766251:example.com"
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              example: {
+                "reason": "this makes me sad"
+              }
+              properties:
+                reason:
+                  type: string
+                  description: The reason the room is being reported.
+        required: true
+      security:
+        - accessTokenQuery: []
+        - accessTokenBearer: []
+      responses:
+        "200":
+          description: The room has been reported successfully.
+          content:
+            application/json:
+              schema:
+                type: object
+              examples:
+                response:
+                  value: {}
+        "404":
+          description: |-
+            The room was not found on the homeserver.
+          content:
+            application/json:
+              schema:
+                $ref: definitions/errors/error.yaml
+              examples:
+                response:
+                  value: {
+                    "errcode": "M_NOT_FOUND",
+                    "error": "The room was not found."
+                  }
+        "429":
+          description: This request was rate-limited.
+          content:
+            application/json:
+              schema:
+                $ref: definitions/errors/rate_limited.yaml
```

Comment:
I'm not sure if this is technically correct since the MSC [strongly recommends](https://github.com/matrix-org/matrix-spec-proposals/pull/4151/files#diff-0d9364264e36813883f313167d1b53b1bbaeb4ad6e687bfaba8aea6c16b2662dR105) but doesn't require rate limiting.

Reply:
You wouldn't be able to rate-limit this endpoint without `429` being defined, so I believe that the MSC intended for such an error code to be spec'd. Adding the error code here does not require that the homeserver send it - just that the client must handle it should it come up. 

An MSC can't require implementations to add rate-limiting, only recommend they do.

Reply:
Ah, thanks for explaining. That makes perfect sense.